### PR TITLE
[Chore] Fix tests migrated to Chainsaw

### PR DESCRIPTION
**Pull request by IshwarKanse**
**Description:** <Describe what has changed.>
* Update tests/e2e-openshift/Dockerfile to install Chainsaw.
* Fix the tests migrated to Chainsaw.

### CODE REVIEW

File: tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml
Comment (IshwarKanse) [inline tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:7]: This was changed as when the annotation is applied by template, the annotation openshift.io/sa.scc.mcs is missing. 

`chainsaw-becoming-lab": unable to find annotation openshift.io/sa.scc.mcs`

When applied by kubectl, the annotation with the right value is added.

**Comment by IshwarKanse** [comment]
```
Tested on OpenShift 4.15:
--- PASS: chainsaw (0.00s)
    --- PASS: chainsaw/smoke-pod-labels (12.40s)
    --- PASS: chainsaw/autoscale (24.61s)
    --- PASS: chainsaw/smoke-targetallocator (20.21s)
    --- PASS: chainsaw/smoke-statefulset (10.73s)
    --- PASS: chainsaw/smoke-simplest (9.95s)
    --- PASS: chainsaw/statefulset-features (42.78s)
    --- PASS: chainsaw/smoke-shareprocessnamespace (9.19s)
    --- PASS: chainsaw/targetallocator-prometheuscr (62.27s)
    --- PASS: chainsaw/smoke-restarting-deployment (12.04s)
    --- PASS: chainsaw/smoke-sidecar-other-namespace (44.39s)
    --- PASS: chainsaw/smoke-sidecar (43.76s)
    --- PASS: chainsaw/instrumentation-python (56.04s)
    --- PASS: chainsaw/create-sm-prometheus-exporters (46.16s)
    --- PASS: chainsaw/target-allocator (10.52s)
    --- PASS: chainsaw/targetallocator-features (68.10s)
    --- PASS: chainsaw/pdb (9.12s)
    --- PASS: chainsaw/route (15.38s)
    --- PASS: chainsaw/targetallocator-kubernetessd (104.58s)
    --- PASS: chainsaw/create-pm-prometheus-exporters (59.58s)
    --- PASS: chainsaw/monitoring (35.83s)
    --- PASS: chainsaw/otlp-metrics-traces (55.93s)
    --- PASS: chainsaw/opampbridge (13.11s)
    --- PASS: chainsaw/multi-cluster (54.55s)
    --- PASS: chainsaw/multiple-configmaps (12.31s)
    --- PASS: chainsaw/smoke-pod-annotations (9.62s)
    --- PASS: chainsaw/smoke-init-containers (16.86s)
    --- PASS: chainsaw/instrumentation-python-multicontainer (42.37s)
    --- PASS: chainsaw/instrumentation-sdk (54.71s)
    --- PASS: chainsaw/smoke-daemonset (11.12s)
    --- PASS: chainsaw/instrumentation-java-multicontainer (16.66s)
    --- PASS: chainsaw/prometheus-config-validation (26.93s)
    --- PASS: chainsaw/instrumentation-nodejs-multicontainer (21.12s)
    --- PASS: chainsaw/instrumentation-nodejs (16.80s)
    --- PASS: chainsaw/instrumentation-nginx-contnr-secctx (17.32s)
    --- PASS: chainsaw/instrumentation-nginx-multicontainer (19.67s)
    --- PASS: chainsaw/instrumentation-nginx (18.10s)
    --- PASS: chainsaw/instrumentation-java-other-ns (23.06s)
    --- PASS: chainsaw/managed-reconcile (15.32s)
    --- PASS: chainsaw/kafka (160.94s)
    --- PASS: chainsaw/ingress (10.97s)
    --- PASS: chainsaw/daemonset-features (17.13s)
    --- PASS: chainsaw/instrumentation-java (15.33s)
    --- PASS: chainsaw/instrumentation-dotnet-multicontainer (17.07s)
    --- SKIP: chainsaw/instrumentation-apache-multicontainer (0.00s)
    --- PASS: chainsaw/ingress-subdomains (51.95s)
    --- PASS: chainsaw/instrumentation-dotnet (14.54s)
    --- PASS: chainsaw/instrumentation-dotnet-musl (14.99s)
    --- PASS: chainsaw/instrumentation-apache-httpd (16.45s)
    --- PASS: chainsaw/instrumentation-go (56.64s)
PASS
Tests Summary...
- Passed  tests 48
- Failed  tests 0
- Skipped tests 1
Done.
```

File: tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml
Comment (eddycharly) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:7]: Things behave differently when the ns is annotated than when the namespace is created with the annotation already set ?

Comment (IshwarKanse) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:7]: Yes, with the CLI the annotation annotation openshift.io/sa.scc.mcs with the correct value is added.

**Review by iblancasa — APPROVED** [review]

File: tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml
Comment (eddycharly) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:7]: I don't know much about openshift but it sounds strange to me that things behave differently when the annotation is added afterward 🤷

Comment (IshwarKanse) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:7]: For the time being we can get this merged so that the tests do not fail on OpenShift. I'm consulting with our OpenShift team on this and will try to find a way to get it working with the template.

Comment (eddycharly) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:7]: Sure, i was just curious :)

File: tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml
Comment (swiatekm-sumo) [inline tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:25]: Why are we changing this? Does the current form not work?

Comment (IshwarKanse) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:25]: So when a new project or namespace in created in OpenShift, the openshift.io/sa.scc.mcs automatically added with other scc annotations. If we set only the uid and group scc annotation via a resource file template, the openshift.io/sa.scc.mcs is not added with other required annotations and we get error in the deployments. 

`chainsaw-becoming-lab": unable to find annotation openshift.io/sa.scc.mcs`

So instead we create the namespace with the annotations auto added and only modify the uid and group scc ones which are required.

Comment (swiatekm-sumo) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml:25]: We should add comments explaining this, then. It's far from obvious, and operator CI doesn't check for OpenShift breakages.

Comment (IshwarKanse) [reply tests/e2e-instrumentation/instrumentation-apache-httpd/chainsaw-test.yaml]: @swiatekm-sumo I added comment to explain why the change is needed.

**Review by swiatekm-sumo — APPROVED** [review]
LGTM, but I'm not an OpenShift expert, would appreciate review from another maintainer of these tests.

**Comment by IshwarKanse** [comment]
@frzifus Need your approval on the PR.

**Review by frzifus — APPROVED** [review]

**Comment by frzifus** [comment]
Seems the e2e test is failing:
```
 === NAME  chainsaw/targetallocator-prometheuscr
    l.go:53: | 15:51:52 | targetallocator-prometheuscr | step-00  | APPLY     | ERROR | rbac.authorization.k8s.io/v1/ClusterRoleBinding @ (join('-', ['ta', $namespace]))
        === ERROR
        metadata.name: Internal error: SyntaxError: Unexpected token at the end of the expression: TOKStringLiteral
=== NAME  chainsaw/targetallocator-features
    l.go:53: | 15:51:52 | targetallocator-features | step-00  | APPLY     | ERROR | rbac.authorization.k8s.io/v1/ClusterRoleBinding @ (join('-', ['default-view', $namespace]))
        === ERROR
        metadata.name: Internal error: SyntaxError: Unexpected token at the end of the expression: TOKStringLiteral
 ```

**Comment by IshwarKanse** [comment]
@frzifus Seems to be related to the templating. Its flaking as it passed in the earlier run. I'll look into it after this gets merged as its not related to the changes in this PR.

**Comment by eddycharly** [comment]
It looks like the templating is not applied 🤔

**Comment by eddycharly** [comment]
Nope, that's a flake, i can reproduce it :(

**Comment by frzifus** [comment]
I triggered a 3rd re-run. Lets see, but if thats not a behavior on `main` I think we should fix it first.

cc @jaronoff97

**Comment by eddycharly** [comment]
It’s definitely a bug in chainsaw. Something goes wrong with parsing expressions concurrently.

**Comment by eddycharly** [comment]
Ok, found it, will be fixed in the next version (hopefully tomorrow).

**Comment by eddycharly** [comment]
For reference https://github.com/kyverno/kyverno-json/pull/312 🙈

**Comment by eddycharly** [comment]
@IshwarKanse i fixed the issue, i'm going to cut `v0.1.5-beta.1` if you want to try it (one PR is missing and will work on the docs, `v0.1.5` should be available tomorrow).

**Comment by IshwarKanse** [comment]
@eddycharly The Github action needs to be updated. I've bumped the version in this PR.

**Comment by eddycharly** [comment]
I'm cutting `v0.1.6` with the fix.

**Comment by eddycharly** [comment]
@IshwarKanse `v0.1.6` is available.

**Comment by eddycharly** [comment]
@IshwarKanse i knew multi cluster support was a risky feature. Thanks for your patience!

**Comment by IshwarKanse** [comment]
@frzifus @jaronoff97 Thanks to @eddycharly ,the Chainsaw bugs have now been resolved and the CI is green. The PR is good to merge.

**Pull request merged**
